### PR TITLE
changed health checks metadata structure

### PR DESCRIPTION
**Pull request by ashleyschuett**

### CODE REVIEW

File: lib/applications.js [hunk 1]
```diff
@@ -241,15 +241,29 @@ class Applications {
...
+                            const enforce_health_check_exists = {
+                                tags: {
+                                    metadata : {
+                                        health_checks : {}
                                     }
                                 }
                             };
-                            container.tags = _.defaultsDeep(clear_health_check, container.tags);
+
+                            _.merge(container, enforce_health_check_exists);
```
Comment (normanjoyner) [inline lib/applications.js]: is this even needed if we always explicitly clear the object?

File: lib/applications.js [hunk 2]
```diff
@@ -260,6 +274,11 @@ class Applications {
             },
             (configuration, callback) => {
                 configuration = _.omit(configuration, 'id');
+
+                if(_.has(container, 'health_checks')) {
```
Comment (normanjoyner) [inline lib/applications.js]: we don't need a conditional here, it can be removed. do it. just do it.

do it

File: lib/applications.js [hunk 3]
```diff
@@ -241,15 +241,29 @@ class Applications {
...
+
+                            _.merge(container, enforce_health_check_exists);
+                            // clear container health checks results from metadata
+                            // in case it has data from old application health check
+                            container.tags.metadata.health_checks = {};
+
+                            _.each(application.health_checks, (health_check) => {
+                                container.tags.metadata.health_checks[health_check.id] = clear_health_check;
+                            });
+                        } else if(container && container.tags && container.tags.metadata && container.tags.metadata.health_checks) {
```
Comment (normanjoyner) [inline lib/applications.js]: why don't we delete the health_check metadata when there are no parent health checks?

File: lib/applications.js (hunk 1 above)
Comment (ashleyschuett) [inline lib/applications.js]: yes it is, on container create at this point the container is `{}`

Comment (normanjoyner) [inline lib/applications.js]: okay 👍

**Pull request merged**